crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -434,6 +434,9 @@ async fn c8y_mapper_syncs_pending_alarms_on_startup() {
     )
     .await;
 
+    // wait for messages to get synced properly before aborting mapper
+    tokio::time::sleep(Duration::from_millis(300)).await;
```

Comment:
This delay may not be enough on lower resource devices, is there a way to wait for all msgs ack and then continue?

Reply:
Rather than waiting some time to give a chance for the messages to be processed, it would be better to improve the mapper to gracefully handle SIGTERM.

=> I will create a ticket for that.